### ISSUE TITLE
SSE: message granularity

### ISSUE THREAD

**Issue by leslie-leder** [ReactiveX/RxNetty#531]
In issue #530 I provided a scala-test snippet with Http SSE response like so:     

```
"\n" +
"data: message1" +
"\n" +
"data: message2" +
"\n" +
"data: message3" +
"\n"
```

, which actually means: 

```
data: message1
data: message2
data: message3
```

This one should result in one message but it is decoded in two messages (the first is somehow discarded). 

Using proper data (I actually intended 3 messages) like so:

```
"\n" +
"data: message1\n" +
"\n" +
"data: message2\n" +
"\n" +
"data: message3\n" +
"\n"
```

, which actually means: 

```
data: message1

data: message2

data: message3
```

produces exactly the same output. 2 messages but it should be 3 messages.

**Comment by NiteshKant** [comment]
@leslie-leder I think this is a duplicate for issue #530 as the only issue here is the existence of leading `\n` before the first data element. If you remove the leading `\n`, the client works as expected. Am I missing something?

**Comment by leslie-leder** [comment]
I'm uncertain about this. I just rephrase my observation. 

If the http-server sends the  following chunk: 

```
data: line1-message1
data: line2-message1
data: line3-message1
```

I would expect to get one event of type ServerSentEvent.Type.Data with 3 lines content like so:

```
line1-message1
line2-message1
line3-message1
```

Whereas if the http-server sends the  following chunk: 

```
data: line1-message1

data: line1-message2

data: line1-message3
```

I think 3 separate messages would be right.

My observation was that in both cases 3 messages were emitted. Unfortunately I cannot concatenate them because I cannot decide whether they belong together or not. The message delimiter - the empty line - is not available anymore.

**Comment by NiteshKant** [comment]
Closed via #543

**Comment by leslie-leder** [comment]
With version 0.5.2-rc.5 this is still an issue. The following content should result in two SSEs but emits three events.

```
data: Greetings
data: Mortal.

data: Are you ready to die?
```

According to the spec I would expect: 

Event 1: 

    Greetings
    Mortal

Event 2: 

    Are you ready to die?

The actual result is:

Event 1: 

    Greetings

Event 2:

    Mortal

Event 3: 

    Are you ready to die?

see below quote from the spec https://www.w3.org/TR/eventsource/

> On the server-side, the script ("updates.cgi" in this case) sends messages in the following form, with the text/event-stream MIME type:
> 
> data: This is the first message.
> 
> data: This is the second message, it
> data: has two lines.
> 
> data: This is the third message.
